prusti-interface/src/specs/checker.rs

```diff
@@ -0,0 +1,118 @@
+use log::{info, debug};
+use rustc_middle::{
+    hir::map::Map,
+    ty::TyCtxt,
+};
+use rustc_span::{Span, MultiSpan};
+use rustc_hir::{
+    self as hir,
+    intravisit::{self, Visitor},
+    itemlikevisit::ItemLikeVisitor,
+    def_id::{DefId, LocalDefId},
+};
+
+use std::collections::HashMap;
+
+use crate::{
+    environment::Environment,
+    utils::has_prusti_attr,
+    PrustiError,
+};
+
+
+pub struct SpecChecker<'tcx> {
+    tcx: TyCtxt<'tcx>,
+
+    predicates: HashMap<DefId, Span>,
```

Comment:
Could you add a `///` comment for the `predicates` field and the `SpecChecker` type?

Reply:
- [x] done in 34cd995b